pkg/client/dtclient/settings_client.go

```diff
@@ -247,6 +257,76 @@ func (d *DynatraceClient) upsertSettings(ctx context.Context, obj SettingsObject
 	return entity, nil
 }
 
+func isSameValueOfConstraint(key string, c1 []byte, c2 []byte) (bool, error) {
+	u := make(map[string]any)
+	if err := json.Unmarshal(c1, &u); err != nil {
+		return false, fmt.Errorf("failed to unmarshal data for key %q: %w", key, err)
+	}
+	v1 := u[key]
+
+	if err := json.Unmarshal(c2, &u); err != nil {
+		return false, fmt.Errorf("failed to unmarshal data for key %q: %w", key, err)
+	}
+	v2 := u[key]
+
+	return reflect.DeepEqual(v1, v2), nil
+}
+
+func (d *DynatraceClient) findObjectWithMatchingConstraints(ctx context.Context, source SettingsObject) (*DownloadSettingsObject, error) {
+	constraints, err := d.fetchSchemasConstraints(ctx, source.SchemaId)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get details for %q schema: %w", source.SchemaId, err)
+	}
+
+	if len(constraints.UniqueProperties) == 0 {
+		return nil, nil
+	}
+
+	objects, err := d.listSettings(ctx, source.SchemaId, ListSettingsOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("unable to get existing settings objects for %q schema: %w", source.SchemaId, err)
+	}
+
+	target, err := findObjectWithSameConstraints(constraints, source, objects)
+	if err != nil {
+		return nil, err
+	}
+	return target, nil
+}
+
+func findObjectWithSameConstraints(schema SchemaConstraints, source SettingsObject, objects []DownloadSettingsObject) (*DownloadSettingsObject, error) {
+	var candidate *DownloadSettingsObject
+	var err error
+	for _, rule := range schema.UniqueProperties {
+		for j, o := range objects {
+			var b bool
+			b, err = areObjectsEqualForTheRule(rule, source, o)
```

Comment:
We overwrite the error in every iteration, we need to create a second one, and append it to the first.